includes/notes/class-wc-payments-notes-set-https-for-checkout.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Set up ensure https on checkout note for WooCommerce inbox.
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+use Automattic\WooCommerce\Admin\Notes\NoteTraits;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class WC_Payments_Notes_Set_Https_For_Checkout
+ */
+class WC_Payments_Notes_Set_Https_For_Checkout {
+	use NoteTraits;
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	const NOTE_NAME = 'wc-payments-notes-set-https-for-checkout';
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	const NOTE_DOCUMENTATION_URL = 'https://docs.woocommerce.com/document/ssl-and-https/#section-7';
+
+	/**
+	 * Get the note.
+	 */
+	public static function get_note() {
+
+		// This note only makes sense if HTTPS is not enforced yet.
+		if ( 'yes' === get_option( 'woocommerce_force_ssl_checkout' ) || wc_site_is_https() ) {
```

Comment:
Looks like we could move this guard condition into `includes/class-wc-payments.php` for better separation of concerns and lazy loading.

Reply:
I am actually divided here. IMHO, putting triggering logic and note contents close to each other would help discoverability and debugging.

I've checked[`NoteTraits`](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Notes/NoteTraits.php) and it has a `can_be_added` method which looks like a better fit of them all. Returning an empty note would make `can_be_added` to return false.

Would moving guard condition to the `can_be_added` method be a viable option?

Reply:
Yes, moving to `can_be_added` is a good alternative, thanks!